finetune_nullspace.sh

```diff
@@ -1,30 +1,90 @@
-SAVE_DIR="/pio/scratch/1/i273233/linear_separability/cpc/gru_level2/cpc_official"
 SPEAKERS="speakers_factorized"
 PHONEMES="phonemes_nullspace"
 SPEAKERS_NULLSPACE="speakers_nullspace"
 
-DIM_INTER=$1
+DATASET_PATH=false
+TRAIN_SET=false
+VALIDATION_SET=false
+CHECKPOINT_PATH=false
+OUTPUT_DIR=false
+DIM_INBETWEEN=false
 FROM_STEP=$SPEAKERS
-if [[ $# -ge 2 ]]; then
-    FROM_STEP=$2
+PHONES_PATH=false
+AUDIO_FORMAT=flac
+
+print_usage() {
+  echo -e "Usage: ./finetune_nullspace.sh"
+  echo -e "\t-d DATASET_PATH (E.g. LIBRISPEECH_DATASET_PATH/train-clean-100)"
+  echo -e "\t-t TRAIN_SET (E.g. LIBRISPEECH_TRAIN_CLEAN_100_TRAIN_SPLIT_FILE_PATH)"
+  echo -e "\t-v VALIDATION_SET (E.g. LIBRISPEECH_TRAIN_CLEAN_100_TEST_SPLIT_FILE_PATH)"
+  echo -e "\t-c CHECKPOINT_PATH"
+  echo -e "\t-o OUTPUT_DIR"
+  echo -e "\t-n DIM_INBETWEEN (Dimension of nullspace will be DIM_EMBEDDING - DIM_INBETWEEN)"
+  echo -e "\t-p PHONES_PATH (Path to the file containing phonemes for the entire dataset)"
+  echo -e "OPTIONAL FLAGS:"
+  echo -e "\t-s FROM_STEP (From which step do you want to start. Order: $SPEAKERS [default] -> $PHONEMES -> $SPEAKERS_NULLSPACE)"
+  echo -e "\t-f audio files format in LibriSpeech dataset (without a dot)"
+}
+
+while getopts 'd:t:v:c:o:n:s:p:f' flag; do
```

Comment:
getopts is stupid and requires a colon after each parameter that is not a flag. So there should be a colon after f: 
`while getopts 'd:t:v:c:o:n:s:p:f:' flag; do`

Reply:
updated; changed also in `eval_abx.py` as you suggested